packages/entity-server/src/routes/hierarchy/types.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { ParamsDictionary, Query } from 'express-serve-static-core';
+import { Request, Response } from 'express';
+import { Context } from '../../types';
+import { EntityFields, EntityType } from '../../models';
+
+interface HierarchyRequestParams extends ParamsDictionary {
+  projectCode: string;
+  entityCode: string;
+}
+
+interface HierarchyRequestQuery extends Query {
+  fields?: string;
+}
+
+export type EntityResponseObject = {
+  [field in keyof EntityFields]?: EntityFields[field];
+};
+
+// eslint-disable-next-line @typescript-eslint/comma-dangle
+export interface HierarchyRequest<
+  P = HierarchyRequestParams,
+  ResBody = EntityResponseObject,
+  ReqBody = Record<string, never>,
```

Comment:
What's the `never` about? I looked it up, but still can't fathom its use here

Reply:
Is it because a `HierarchyRequest` is always a `GET`, so shouldn't have a body...?

Reply:
Yeah it was a bit of a weird thing to do hehe. I googled around and this is what was recommended to me when you want to suggest that a type should be empty. I'm happy to go with anything really since this is just the default setting, and ideally we'd be explicitly setting what the ReqBody should look like per route